src/components/Backdrop/index.js

```diff
@@ -1,90 +1,61 @@
-  ESCAPE: 27
-};
+import { Overlay } from "./index.styles";
 
-const Overlay = styled.div.attrs({
-  className: "container--overlay"
-})`
-  height: 100%;
-  width: 100%;
-  position: fixed;
-  left: 0;
-  top: 0;
-  background-color: rgba(0, 0, 0, 0.75);
-  z-index: 4;
-  display: flex;
-  justify-content: center;
-  align-items: center;
-`;
+export default class Backdrop extends Component {
+  static propTypes = {
+    /* eslint-disable-next-line react/forbid-prop-types */
+    childRef: PropTypes.object.isRequired,
+    children: PropTypes.node.isRequired,
+    overlay: PropTypes.bool,
+    onRequestClose: PropTypes.func
+  };
 
-class Backdrop extends Component {
-  state = {
-    childRef: React.createRef()
+  static defaultProps = {
+    overlay: true,
+    onRequestClose: null
   };
+
   componentDidMount() {
-    document.addEventListener("click", this.handleOutsideClick, false);
-    document.addEventListener("keydown", this.onKeyPress, false);
+    document.addEventListener("click", this.handleOutsideClick);
+    document.addEventListener("keydown", this.onKeyPress);
   }
 
   componentWillUnmount() {
-    document.removeEventListener("click", this.handleOutsideClick, false);
-    document.removeEventListener("keydown", this.onKeyPress, false);
+    document.removeEventListener("click", this.handleOutsideClick);
+    document.removeEventListener("keydown", this.onKeyPress);
   }
 
-  onKeyPress = event => {
-    if (event.keyCode === KEYCODES.ESCAPE) {
-      this.onCloseRequest();
+  onKeyPress = e => {
+    const { onRequestClose } = this.props;
+
+    if (e.keyCode === ESCAPE && onRequestClose) {
+      onRequestClose(e);
     }
   };
 
   handleOutsideClick = e => {
-    const { childRef } = this.state;
-
-    if (childRef && childRef.current) {
-      if (!childRef.current.contains(e.target)) {
-        if (this.onCloseRequest) {
-          this.onCloseRequest();
-        }
-      }
-    } else {
-      this.onCloseRequest();
+    const { childRef, onRequestClose } = this.props;
+
+    if (
+      onRequestClose &&
+      childRef.current &&
+      !childRef.current.contains(e.target)
+    ) {
+      onRequestClose(e);
     }
   };
 
   render() {
-    const { children, ...rest } = this.props;
-    return (
-      <ItemContainerConsumer>
-        {value => {
-          this.onCloseRequest = value ? value.onCloseRequest : () => {};
-          return (
-            <Overlay onClick={this.handleOutsideClick} {...rest}>
-              {children && (
-                <BackdropProvider value={this.state}>
-                  {children}
-                </BackdropProvider>
-              )}
-            </Overlay>
-          );
-        }}
-      </ItemContainerConsumer>
-    );
-  }
-}
-
-Backdrop.defaultProps = {
-  children: null
-};
+    const { children, overlay } = this.props;
 
-Backdrop.propTypes = {
-  children: PropTypes.node
-};
+    if (overlay) {
+      return <Overlay data-testid="overlay">{children}</Overlay>;
```

Comment:
`data-testid`? We shouldn't have any `data-tids` hardcoded in `aurora`.

Reply:
I'll try test it in other way

Reply:
Now I just passing down props

Reply:
Looks like it is still on the branch